### PR TITLE
Implement chanmode params.

### PR DESCRIPTION
**Pull request by Timer**
Opening this pull request to gain feedback and thoughts.
Once approved, I'll write up a set of tests to ensure it all works and no bugs will arise from the changes.

### CODE REVIEW

**Comment by Timer** [comment]
Very similar to #145.

File: lib/irc.js [hunk 1]
```diff
@@ -265,39 +265,51 @@ function Client(server, nick, opt) {
...
+                    var eventName = (adding ? '+' : '-') + 'mode';
+                    var supported = self.supported.channel.modes;
+                    var modeArg;
+                    var chanModes = function(mode, param) {
+                        var arr = Array.isArray(param);
                         if (adding) {
-                            if (channel.users[user].indexOf(self.prefixForMode[mode]) === -1)
-                                channel.users[user] += self.prefixForMode[mode];
-
-                            self.emit('+mode', message.args[0], message.nick, mode, user, message);
```
Comment (jirwin) [inline lib/irc.js]: It is important that we still emit these events(`+mode` and `-mode`)

**Comment by jirwin** [comment]
Tests please

**Comment by Timer** [comment]
Added tests.

**Comment by Timer** [comment]
Rebased to remove merge commits, per request of @vBm on IRC.

**Comment by vBm** [comment]
Sorry if it sound as a request, it was just merely a suggestion.
But good job :P

**Comment by jirwin** [comment]
Looks great. Thanks for this! I manually rebased this and merged it in ee5c5502c233bb2b824fbc62bfad6b198fb155a1

**Comment by Timer** [comment]
Cool @jirwin! I would've rebased if you asked.

**Pull request closed without merging**